src/main/java/pl/edu/icm/pl/mxrdr/extension/workflow/XdsOutputImportStep.java

```diff
@@ -0,0 +1,100 @@
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.workflow.WorkflowExecutionContext;
+import edu.harvard.iq.dataverse.workflow.step.Failure;
+import edu.harvard.iq.dataverse.workflow.step.FilesystemAccessingWorkflowStep;
+import edu.harvard.iq.dataverse.workflow.step.Success;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult.Source;
+import io.vavr.control.Try;
+import pl.edu.icm.pl.mxrdr.extension.importer.xds.XdsImporter;
+import pl.edu.icm.pl.mxrdr.extension.importer.xds.XdsImporterForm;
+
+public class XdsOutputImportStep extends FilesystemAccessingWorkflowStep {
+
+    private static final String WORKFLOW = "WORKFLOW";
+
+    static final String CORRECT_LP = "CORRECT.LP";
+    
+    ImporterRegistry importers;
+
+    private DatasetFieldServiceBean datasetFieldService;
+
+    private DatasetVersionServiceBean datasetVersionService;
+
+
+    public XdsOutputImportStep(Map<String, String> inputParams, DatasetFieldServiceBean datasetFieldService, DatasetVersionServiceBean datasetVersionService, ImporterRegistry importers) {
+        super(inputParams);
+        this.datasetFieldService = datasetFieldService;
+        this.datasetVersionService = datasetVersionService;
+        this.importers = importers;
+    }
+
+    @Override
+    public WorkflowStepResult resume(WorkflowExecutionContext context, Map<String, String> arg1, String arg2) {
+        return null;
+    }
+
+    @Override
+    public void rollback(WorkflowExecutionContext context, Failure failure) {
+    }
+
+    @Override
+    protected Source runInternal(WorkflowExecutionContext context, Path workdirPath) throws Exception {
+        
+        File correctionFile = workdirPath.resolve(CORRECT_LP).toFile();
+        Dataset dataset = context.getDataset();
+        for (MetadataImporter importer:importers.getImporters().values()) {
+            if (importer instanceof XdsImporter) {
+                Map<ImporterFieldKey, Object> inputParametersMap = new HashMap<>();
+                inputParametersMap.put(XdsImporterForm.XDS_FILE, correctionFile);
+                List<ResultField> resultFields = importer.fetchMetadata(inputParametersMap);
+                DatasetVersion editVersion = dataset.getEditVersion();
+                
+                for (ResultField resultField:resultFields) {
+                    if (resultField.getChildren().isEmpty()) {
+                        addField(editVersion, resultField);
+                    } else {
+                        for (ResultField childField:resultField.getChildren()) {
+                            addField(editVersion, childField);
+                        }
+                    }
+                }
+                
+                Try<Dataset> updateDataset = Try.of(() -> datasetVersionService.updateDatasetVersion(editVersion, true));
+
+                if (updateDataset.isSuccess()){
```

Comment:
Handling only successful `Try` result is equivalent of exception swallowing - let's not. Also `Try` is a functional construct, hence it works better when used in functional style.